### PR TITLE
Added XMLDocs

### PR DESCRIPTION
**Pull request by to11mtm**
Added XMLDocs to `ExpressionAttribute`, `FunctionAttribute`, and `PropertyAttribute`

Added XMLDocs to `SQLiteDataProvider` Constructors

### CODE REVIEW

**Comment by to11mtm** [comment]
Related to #2093

File: Source/LinqToDB/Sql/Sql.ExpressionAttribute.cs [hunk 1]
```diff
@@ -11,33 +11,66 @@ namespace LinqToDB
...
 		[PublicAPI]
 		[Serializable]
 		[AttributeUsage(AttributeTargets.Property | AttributeTargets.Method, AllowMultiple = true, Inherited = false)]
 		public class ExpressionAttribute : Attribute
 		{
+			/// <summary>
+			/// Creates an Expression that will be used in SQL,
+			/// in place of the method call decorated by this attribute. 
+			/// </summary>
+			/// <param name="expression">The SQL expression. Use {0},{1}... for parameters given to the method call</param>
```
Comment (MaceWindu) [inline Source/LinqToDB/Sql/Sql.ExpressionAttribute.cs]: Please add missing trailing dots to all statements.

File: Source/LinqToDB/Sql/Sql.ExpressionAttribute.cs [hunk 2]
```diff
@@ -46,19 +79,60 @@ public ExpressionAttribute(string configuration, string expression, params int[]
...
+			/// </summary>
 			public string         Expression       { get; set; }
+			/// <summary>
+			/// The order of Arguments to be passed
+			/// into the function from the method call
+			/// </summary>
 			public int[]          ArgIndices       { get; set; }
+			/// <summary>
+			/// Determines the priority of the expression in evaluation
+			/// </summary>
```
Comment (MaceWindu) [inline Source/LinqToDB/Sql/Sql.ExpressionAttribute.cs]: Makes sense to add reference to https://github.com/linq2db/linq2db/blob/master/Source/LinqToDB/SqlQuery/Precedence.cs class

File: Source/LinqToDB/Sql/Sql.ExpressionAttribute.cs [hunk 3]
```diff
@@ -46,19 +79,60 @@ public ExpressionAttribute(string configuration, string expression, params int[]
...
 			public bool           IsPredicate      { get; set; }
+			/// <summary>
+			/// If true, this expression represents an aggregate result
+			/// Examples would be SUM(),COUNT().
+			/// </summary>
 			public bool           IsAggregate      { get; set; }
+			/// <summary>
+			/// Determines whether the return type should be treated as
+			/// something that can be null.
+			/// </summary>
```
Comment (MaceWindu) [inline Source/LinqToDB/Sql/Sql.ExpressionAttribute.cs]: Maybe add notes that it defines nullability rules to make clear distinction from CanBeNull and specify how this option works with CanBeNull

File: Source/LinqToDB/Sql/Sql.ExpressionAttribute.cs [hunk 4]
```diff
@@ -46,19 +79,60 @@ public ExpressionAttribute(string configuration, string expression, params int[]
...
 			public IsNullableType IsNullable       { get; set; }
 
 			internal  bool? _canBeNull;
+			/// <summary>
+			/// If True, result can be null
```
Comment (MaceWindu) [inline Source/LinqToDB/Sql/Sql.ExpressionAttribute.cs]: I see inconsistence in boolean values use in xml-docs: True, 'true'. In general we use &lt;c&gt;true&lt;/c&gt; in other places.

File: Source/LinqToDB/Sql/Sql.FunctionAttribute.cs [hunk 5]
```diff
@@ -12,41 +12,75 @@ namespace LinqToDB
 
 	partial class Sql
 	{
+		/// <summary>
+		/// Defines an SQL Server side Function with parameters passed in.
```
Comment (MaceWindu) [inline Source/LinqToDB/Sql/Sql.FunctionAttribute.cs]: Shouldn't it be "server-side"?

**Comment by MaceWindu** [comment]
@to11mtm could you update your PR before next thursday release?

**Comment by to11mtm** [comment]
Apologies for delays, I have updated with suggested changes and rebased to current.

**Comment by MaceWindu** [comment]
Thanks!

**Pull request merged**